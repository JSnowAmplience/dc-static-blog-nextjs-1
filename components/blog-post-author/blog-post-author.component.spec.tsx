/* eslint-env jest */
import renderer from 'react-test-renderer';
import BlogPostAuthor from './blog-post-author.component';

describe('BlogPostAuthor', () => {
  test('renders author with an avatar, name and read time', (): void => {
    const props = {
      authors: [
        {
          name: 'Sir Bloblog',
          avatar: {
            _meta: {
              schema: 'https://schema.localhost.com/image.json',
              deliveryId: '9ef5a6d0-5a34-414f-9fd3-1098698c6629',
              name: 'bloblog-avatar'
            },
            image: {
              defaultHost: 'i1-qa.adis.ws',
              endpoint: 'bloblogltd',
              name: 'man-business-suit',
              id: 'f9549d3d-f616-487c-b1ad-fad5152f2c7d',
              _meta: {
                schema: 'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link'
              },
              mediaType: 'i'
            },
            altText: 'The dashing Sir Bloblog avatar',
            src: '//i1-qa.adis.ws/i/bloblogltd/man-business-suit'
          }
        }
      ],
      readTime: 15,
      date: '2019-08-14'
    };
    const component = renderer.create(<BlogPostAuthor {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
