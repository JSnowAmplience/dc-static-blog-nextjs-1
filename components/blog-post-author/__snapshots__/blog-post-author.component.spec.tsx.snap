// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostAuthor renders author with an avatar, name and read time 1`] = `
<React.Fragment>
  <section
    className="jsx-3448078682"
  >
    <div
      className="jsx-3448078682 avatars"
    >
      <React.Fragment>
        <div
          className="jsx-3144626491 avatar"
        >
          <Image
            altText="The dashing Sir Bloblog avatar"
            dynamicImagingOptions={
              Array [
                Object {
                  "h": 72,
                  "scaleFit": "poi",
                  "sm": "c",
                  "w": 72,
                },
              ]
            }
            src="//i1-qa.adis.ws/i/bloblogltd/man-business-suit"
          />
        </div>
        <JSXStyle
          id="3144626491"
        >
          .avatar.jsx-3144626491 img{object-fit:cover;height:52px;width:52px;border-radius:4px;margin-right:20px;}
        </JSXStyle>
      </React.Fragment>
    </div>
    <div
      className="jsx-3448078682 publish-data"
    >
      <div
        className="jsx-3448078682 authors"
      >
        <React.Fragment>
          <div
            className="jsx-940982213 name"
          >
            Sir Bloblog
          </div>
          <JSXStyle
            dynamic={
              Array [
                "#333333",
                "1.125rem",
                "500",
              ]
            }
            id="2651991810"
          >
            .name.__jsx-style-dynamic-selector{color:#333333;font-size:1.125rem;font-weight:500;}
          </JSXStyle>
        </React.Fragment>
      </div>
      <div
        className="jsx-3448078682 publish-metadata"
      >
        <div
          className="jsx-3448078682 date"
        >
          2019-08-14
        </div>
        <div
          className="jsx-3448078682 readTime"
        >
          15
           mins read
        </div>
      </div>
    </div>
  </section>
  <JSXStyle
    dynamic={
      Array [
        "#cccccc",
        "#999999",
        "#999999",
        "1.125rem",
        "400",
        "736px",
      ]
    }
    id="2864599872"
  >
    section.__jsx-style-dynamic-selector{margin-top:75px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;padding-bottom:25px;max-width:740px;border-bottom:1px solid #cccccc;}.date.__jsx-style-dynamic-selector{padding-right:6px;margin-right:6px;border-right:1px solid #999999;}.date.__jsx-style-dynamic-selector,.readTime.__jsx-style-dynamic-selector{color:#999999;font-size:1.125rem;font-weight:400;}.publish-data.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.publish-metadata.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-top:5px;}@media (max-width:736px){section.__jsx-style-dynamic-selector{margin-top:45px;}}
  </JSXStyle>
</React.Fragment>
`;

exports[`BlogPostAuthor renders author without an avatar, but still has name and read time 1`] = `
<React.Fragment>
  <section
    className="jsx-3448078682"
  >
    <div
      className="jsx-3448078682 avatars"
    />
    <div
      className="jsx-3448078682 publish-data"
    >
      <div
        className="jsx-3448078682 authors"
      >
        <React.Fragment>
          <div
            className="jsx-940982213 name"
          >
            Sir Bloblog
          </div>
          <JSXStyle
            dynamic={
              Array [
                "#333333",
                "1.125rem",
                "500",
              ]
            }
            id="2651991810"
          >
            .name.__jsx-style-dynamic-selector{color:#333333;font-size:1.125rem;font-weight:500;}
          </JSXStyle>
        </React.Fragment>
      </div>
      <div
        className="jsx-3448078682 publish-metadata"
      >
        <div
          className="jsx-3448078682 date"
        >
          2019-08-14
        </div>
        <div
          className="jsx-3448078682 readTime"
        >
          15
           mins read
        </div>
      </div>
    </div>
  </section>
  <JSXStyle
    dynamic={
      Array [
        "#cccccc",
        "#999999",
        "#999999",
        "1.125rem",
        "400",
        "736px",
      ]
    }
    id="2864599872"
  >
    section.__jsx-style-dynamic-selector{margin-top:75px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;padding-bottom:25px;max-width:740px;border-bottom:1px solid #cccccc;}.date.__jsx-style-dynamic-selector{padding-right:6px;margin-right:6px;border-right:1px solid #999999;}.date.__jsx-style-dynamic-selector,.readTime.__jsx-style-dynamic-selector{color:#999999;font-size:1.125rem;font-weight:400;}.publish-data.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.publish-metadata.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-top:5px;}@media (max-width:736px){section.__jsx-style-dynamic-selector{margin-top:45px;}}
  </JSXStyle>
</React.Fragment>
`;
